### ISSUE TITLE
`vue/component-tags-order` should enforce order between `<script>` and `<script setup>`

### ISSUE THREAD

**Issue by achaphiv** [vuejs/eslint-plugin-vue#1819]
**What rule do you want to change?**
vue/component-tags-order

**Does this change cause the rule to produce more or fewer warnings?**
More.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New configuration option. 

Possibly the new default?

**Example**

```vue
<script setup>
  // ...
</script>

<script>
  // ...
</script>
```

**What does the rule currently do for this code?**
Nothing.

**What will the rule do after it's changed?**

Move `<script setup>` below `<script>`.

```vue
<script>
  // ...
</script>

<script setup>
  // ...
</script>
```
